### DISCUSSION TITLE
Support css variables in style designer

### DISCUSSION THREAD

**Discussion by lofcz** [GrapesJS/grapesjs#4132] [Ideas]
Persisted component, background color is set to some css variable. Works fine:
![image](https://user-images.githubusercontent.com/10260230/152866963-7eb4d678-d97f-4b8c-9ddc-ff4dec61fbae.png)

When input is edited and result is a valid usage of `var()` style manager incorrectly treats it as a invalid value:
![image](https://user-images.githubusercontent.com/10260230/152866849-207dfa87-2903-4695-9a43-943f13adfd61.png)

**Comment by artf** [comment]
Yeah right now all the input values are processed by an internal validator, so for many types something like `var(*` is not a valid value.
Probably I'll need to find a way to extend that kind of behavior.

**Reply by lofcz** [reply]
It might be worth to allow for a custom handler to override the default impl. For example I could then write just `--my-var` and transform that to `var(--my-var)`. But variables with valid notation should probably be supported by default.

**Reply by artf** [reply]
> But variables with valid notation should probably be supported by default.

Yeah, I agree.

**Comment by lexoyo** [comment]
Hello
I'm thinking we should have a plugin to support css variables or design tokens
Would you agree ? Will that be in the core @artf ?
